packages/sdk-js/src/index.ts

```diff
@@ -749,150 +838,248 @@ class Client extends BaseClient {
   async runNodeWithInputs(
-    request: RunNodeWithInputsRequest,
+    {
+      nodeType,
+      nodeConfig,
+      inputVariables = {},
+      walletAddress,
+    }: RunNodeWithInputsRequest,
     options?: RequestOptions
   ): Promise<RunNodeWithInputsResponse> {
-    let workflowId: string | null = null;
-    let dedicatedWallet: any = null;
-    
     try {
-      // Generate a numeric salt that can be parsed by math.ParseBig256
-      // Use timestamp + random number to ensure uniqueness
-      const timestamp = Date.now();
-      const randomNum = Math.floor(Math.random() * 1000000000); // 9 digits
-      const salt = `${timestamp}${randomNum.toString().padStart(9, '0')}`;
-      dedicatedWallet = await this.getWallet({ salt });
-      
-      // Create a temporary node using NodeFactory patterns
-      const nodeId = `temp_${Date.now()}`;
-      const node = this.createNodeFromConfig(request.nodeType, request.nodeConfig, nodeId);
+      // Special handling for blockTrigger nodes - they should reject input variables
+      if (nodeType === 'blockTrigger' && inputVariables && Object.keys(inputVariables).length > 0) {
+        const variableNames = Object.keys(inputVariables).join(', ');
+        return {
+          success: false,
+          data: null, // Now allowed by updated interface
+          error: `blockTrigger nodes do not accept input variables. Received: ${variableNames}`,
+          nodeId: ''
+        };
+      }
       
-      // Create minimal workflow with just this node
+      // Map the string nodeType to the enum NodeType
+      const mappedNodeType = this.mapNodeTypeString(nodeType);
+      if (!mappedNodeType) {
+        return {
+          success: false,
+          error: `Unsupported node type: ${nodeType}`,
+        };
+      }
+
+      // Create a node from the config
+      const node = this.createNodeFromConfig(
+        mappedNodeType,
+        nodeConfig,
+        `node_${Date.now()}`
+      );
+
+      // Create a temporary workflow with the node
       const triggerId = `trigger_${Date.now()}`;
-      const tempWorkflow = this.createTempWorkflowForNode(node, triggerId, request.inputVariables, dedicatedWallet.address);
-      
-      // Submit and immediately trigger the workflow
-      workflowId = await this.submitWorkflow(tempWorkflow);
-      
-      const execution = await this.triggerWorkflow({
-        id: workflowId,
-        reason: { type: TriggerType.Manual },
-        isBlocking: true
-      });
-      
-      return {
-        success: execution.status === ExecutionStatus.FINISHED,
-        data: this.extractNodeExecutionData(execution),
-        executionId: execution.executionId,
-        nodeId: nodeId
-      };
-    } catch (error) {
+      const workflow = this.createTempWorkflowForNode(
+        node,
+        triggerId,
+        inputVariables,
+        walletAddress || ""
+      );
+
+      // Submit the workflow
+      const workflowId = await this.submitWorkflow(workflow, options);
```

Comment:
No workflow should involve in this function at all. Simply pass request through to the server.